composer/callbacks/checkpoint_saver.py

```diff
@@ -59,14 +62,17 @@ def save_interval(state: State, event: Event):
         if elapsed_duration >= 1.0:
             return True
 
-        if save_event == Event.EPOCH_CHECKPOINT:
-            count = state.timestamp.epoch
-        elif save_event == Event.BATCH_CHECKPOINT:
-            count = state.timestamp.batch
+        assert state.previous_timestamp is not None
```

Comment:
Return false if None? Otherwise, if algo calls before it's set, you get nasty trace